receiver/kafkareceiver/avro_unmarshaler.go

```
@@ -0,0 +1,159 @@
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	resourceAttributes := resourceLogs.Resource().Attributes()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	logRecord := scopeLogs.LogRecords().AppendEmpty()
+	logRecordAttributes := logRecord.Attributes()
+
+	for sourceField, target := range a.mapping {
+		switch {
+		case isResourceAttribute(target):
+			targetField := strings.TrimPrefix(target, resourceAttributesPrefix)
+			mapAttribute(avroLog, sourceField, &resourceAttributes, targetField)
+		case isLogRecordAttribute(target):
+			targetField := strings.TrimPrefix(target, logRecordAttributesPrefix)
+			mapAttribute(avroLog, sourceField, &logRecordAttributes, targetField)
+		case target == bodyField:
+			logRecord.Body().SetStr(avroLog[sourceField].(string))
+		case target == timestampField:
+			timestamp := avroLog[timestampField].(time.Time)
+			logRecord.SetTimestamp(pcommon.NewTimestampFromTime(timestamp))
+		case target == severityTextField:
+			logRecord.SetSeverityText(avroLog[sourceField].(string))
+		case target == severityNumberField:
+			severityNumber, ok := avroLog[sourceField].(int32)
+			if !ok {
+				severityNumber = 0
+			}
+			logRecord.SetSeverityNumber(plog.SeverityNumber(severityNumber))
+		}
+	}
+
+	return logs
+}
+
+func isResourceAttribute(targetField string) bool {
+	return strings.HasPrefix(targetField, resourceAttributesPrefix)
+}
+
+func isLogRecordAttribute(targetField string) bool {
+	return strings.HasPrefix(targetField, logRecordAttributesPrefix)
+}
+
+func mapAttribute(avroLog map[string]interface{}, sourceField string, attributes *pcommon.Map, targetField string) {
+	sourceValue := avroLog[sourceField]
+	if sourceValue != nil {
+		setAttribute(attributes, targetField, sourceValue)
+	}
+}
+
+func setAttribute(attributes *pcommon.Map, key string, value interface{}) {
+	attribute := attributes.PutEmpty(key)
+	_ = attribute.FromRaw(value)
```

Is this the error being ignored?

---

Sorry, this should have read, Why is this error be ignored?